hbase-client/src/main/java/org/apache/hadoop/hbase/quotas/QuotaTableUtil.java

```diff
@@ -628,6 +628,34 @@ static Put createPutForNamespaceSnapshotSize(String namespace, long size) {
     }
   }
 
+  /**
+   * Remove table usage snapshots (u:p columns) for the namespace passed
+   * @param connection connection to re-use
+   * @param namespace the namespace to fetch the list of table usage snapshots
+   */
+  static void deleteTableUsageSnapshotsForNamespace(Connection connection, String namespace)
+    throws IOException {
+    Scan s = new Scan();
+    //Get rows for all tables in namespace
+    s.setRowPrefixFilter(Bytes.toBytes("t." + namespace));
```

Comment:
need the namespace delimiter at the end of your prefix filter, otherwise you'll over-match

e.g. deleting usage for the namespace 'foo' would also end up matching and deleting for the namespace 'foobar'. Making it 't.foo:' should be good enough.

Reply:
Bonus points if you write a test which demonstrates the bug before you fix it :)

Reply:
Thanks Josh for the good catch!

Reply:
I added a test to check for the bug, in my test I create 2 namespaces with similar names and tested that the code only deletes the usage columns for the tables in the deleted namespace.